packages/compass-saved-aggregations-queries/src/components/edit-item-modal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Modal,
+  H3,
+  TextInput,
+  spacing,
+  Button,
+  css,
+} from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import type { MapDispatchToProps, MapStateToProps } from 'react-redux';
+import type { RootState } from '../stores';
+import type { UpdateItemAttributes } from '../stores/edit-item';
+import type { Item } from '../stores/aggregations-queries-items';
+import { cancelEditItem, updateItem } from '../stores/edit-item';
+
+type EditItemModalProps = {
+  isModalOpen: boolean;
+  item?: Pick<Item, 'id' | 'name' | 'type'>;
+  onSubmit(id: string, attributes: UpdateItemAttributes): void;
+  onCancel: () => void;
+};
+
+const formTitleStyles = css({
+  marginBottom: spacing[5],
+});
+
+const formFooterStyles = css({
+  marginTop: spacing[5],
+  display: 'flex',
+  justifyContent: 'flex-end',
+});
+
+const cancelButtonStyles = css({
+  marginRight: spacing[2],
+});
+
+const EditItemModal: React.FunctionComponent<EditItemModalProps> = ({
+  isModalOpen,
+  item,
+  onSubmit,
+  onCancel,
+}) => {
+  const [name, setName] = useState(item?.name ?? '');
+  useEffect(() => {
+    setName(item?.name ?? '');
+  }, [item]);
+
+  if (!item) {
+    return <></>;
+  }
+
+  const isSubmitDisabled = () => {
+    return !name || name === item.name;
+  };
+
+  const onSubmitForm = () => {
+    if (!isSubmitDisabled()) {
+      onSubmit(item.id, { name });
+    }
+  };
+
+  return (
+    <Modal setOpen={onCancel} open={isModalOpen} data-testid="edit-item-modal">
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          onSubmitForm();
+        }}
+      >
+        <H3 className={formTitleStyles}>{`Edit ${item.type}`}</H3>
+        <TextInput
+          aria-label="Name"
+          label="Name"
+          name="name"
+          value={name}
+          onChange={(event) => {
+            setName(event.target.value);
+          }}
+        />
+        <div className={formFooterStyles}>
+          <Button
+            className={cancelButtonStyles}
+            variant="default"
+            onClick={onCancel}
+          >
+            Cancel
+          </Button>
+          <Button
+            disabled={isSubmitDisabled()}
+            variant="primary"
+            onClick={(event) => {
+              event.preventDefault();
+              onSubmitForm();
+            }}
```

Comment:
IIRC even without the onClick handler and preventing the default it should just trigger a onSubmit on a form and so we don't need to call it here additionally. I'm guessing though leafygreen doesn't allow to omit the onClick prop on the component?